base/compiler/ssair/inlining.jl

```diff
@@ -1280,19 +1351,30 @@ function handle_const_call!(
             # No applicable methods; try next union split
             handled_all_cases = false
             continue
+        else
+            if length(meth) == 1 && only_method !== false
+                if only_method === nothing
+                    only_method = meth[1].method
+                elseif only_method !== meth[1].method
+                    only_method = false
+                end
+            else
+                only_method = false
+            end
         end
         for match in meth
             j += 1
             result = results[j]
             any_fully_covered |= match.fully_covers
+            check_sparams = isa(only_method, Bool)
```

Comment:
Isn't `only_method` being computed within this loop and so it's not fixed at this point?
I think we can use that after we exit this loop.

Reply:
I think it is an OK usage here in terms of whether `meth` itself has >1 match or not, which should be all we care about for `validate_sparams` within `handle_match`